pkg/lightning/errormanager/errormanager.go

```diff
@@ -686,75 +802,95 @@ func (em *ErrorManager) ReplaceConflictKeys(
+							continue
+						}
+
+						// if the KV pair is contained in mustKeepKvPairs, we cannot delete it
+						// if not, delete the KV pair
+						if mustKeepKvPairs != nil {
+							isContained := slices.ContainsFunc(mustKeepKvPairs.Pairs, func(mustKeepKvPair common.KvPair) bool {
+								return bytes.Equal(mustKeepKvPair.Key, kvPair.Key) && bytes.Equal(mustKeepKvPair.Val, kvPair.Val)
+							})
+							if isContained {
+								continue
+							}
+						}
 
-			// find out all the KV pairs that are contained in the data KV
-			kvPairs := encoder.SessionCtx.TakeKvPairs()
-			for _, kvPair := range kvPairs.Pairs {
-				em.logger.Debug("got encoded KV",
-					logutil.Key("key", kvPair.Key),
-					zap.Binary("value", kvPair.Val))
-				kvLatestValue, err := fnGetLatest(gCtx, kvPair.Key)
-				if tikverr.IsErrNotFound(err) {
-					continue
+						handleKeys = append(handleKeys, kvPair.Key)
+					}
 				}
-				if err != nil {
+				if err := dataKvRows.Err(); err != nil {
 					return errors.Trace(err)
 				}
-
-				// if the value of the KV pair is not equal to the latest value of the key of the KV pair
-				// that means the value of the KV pair has been overwritten, so it needs no extra operation
-				if !bytes.Equal(kvLatestValue, kvPair.Val) {
-					continue
+				if err := dataKvRows.Close(); err != nil {
+					return errors.Trace(err)
 				}
-
-				// if the KV pair is contained in mustKeepKvPairs, we cannot delete it
-				// if not, delete the KV pair
-				if mustKeepKvPairs != nil {
-					isContained := slices.ContainsFunc(mustKeepKvPairs.Pairs, func(mustKeepKvPair common.KvPair) bool {
-						return bytes.Equal(mustKeepKvPair.Key, kvPair.Key) && bytes.Equal(mustKeepKvPair.Val, kvPair.Val)
-					})
-					if isContained {
-						continue
-					}
+				if len(handleKeys) == 0 {
+					break
 				}
-
-				if err := fnDeleteKey(gCtx, kvPair.Key); err != nil {
+				if err := fnDeleteKeys(dataGCtx, handleKeys); err != nil {
 					return errors.Trace(err)
 				}
+				start = lastRowID + 1
+				// If the remaining tasks cannot be processed at once, split the task
+				// into two subtasks and send one of them to the other idle worker if possible.
+				if end-start > rowLimit {
+					mid := start + (end-start)/2
+					dataTaskWg.Add(1)
+					select {
+					case dataTaskCh <- [2]int64{mid, end}:
+						end = mid
+					default:
+						dataTaskWg.Done()
+					}
+				}
+				handleKeys = handleKeys[:0]
 			}
+			return nil
+		})
+	}
+	if err := dataG.Wait(); err != nil {
+		return errors.Trace(err)
+	}
+
+	for {
+		nullDataRows, err := em.db.QueryContext(
+			ctx, common.SprintfWithIdentifiers(selectNullDataRows, em.schema))
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		hasRow := false
+		for nullDataRows.Next() {
+			hasRow = true
 		}
-		if err := dataKvRows.Err(); err != nil {
+		if err := nullDataRows.Err(); err != nil {
 			return errors.Trace(err)
 		}
+		if err := nullDataRows.Close(); err != nil {
+			return errors.Trace(err)
+		}
```

Comment:
Since the `conflict_error_v2` table is for users to check the conflicting data, here the rows to be deleted are just for our algorithm to use, they are the rows that needs to be deleted, but not the originally conflicting data, so I think it is necessary to delete them.
I can add an index on `key_data`, which could help improve the deleting speed.